src/process/pagemap.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{FileWrapper, ProcResult};
+
+use bitflags::bitflags;
+use std::{
+    io::{BufReader, Read, Seek, SeekFrom},
+    mem::size_of,
+    ops::{Bound, RangeBounds},
+};
+
+const fn genmask(high: usize, low: usize) -> u64 {
+    let mask_bits = size_of::<u64>() * 8;
+    (!0 - (1 << low) + 1) & (!0 >> (mask_bits - 1 - high))
+}
+
+// source: include/linux/swap.h
+const MAX_SWAPFILES_SHIFT: usize = 5;
+
+// source: fs/proc/task_mmu.c
+bitflags! {
+    /// Represents the fields and flags in a page table entry for a swapped page.
+    pub struct SwapPageFlags: u64 {
+        /// Swap type if swapped
+        const SWAP_TYPE = genmask(MAX_SWAPFILES_SHIFT - 1, 0);
+        /// Swap offset if swapped
+        const SWAP_OFFSET = genmask(54, MAX_SWAPFILES_SHIFT);
```

Comment:
If I'm understanding this correctly, these two entries are bitmasks, and the only reason they are declared like this is so that the call to `from_bits_truncate` will not discard these bits.  Is that right?

If so, then I'm wondering if we can improve the doc-comment on these to indicate that a user would never need to test to see if `SWAP_TYPE` or `SWAP_OFFSET` are set (but instead should use `get_swap_type()` and `get_swap_offset()`.

I guess another option might be to use `from_bits_unchecked` and store these masks outside of the bitflags (though I'd be worried about certain bitflag options later losing these bits).

Can these marks be marked as `#[doc(hidden)]` to reduce confusion for the user?

Reply:
See comment below.